File: app/src/main/java/com/theone/music/ui/fragment/SettingFragment.java
package com.theone.music.ui.fragment;//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛

import static com.qmuiteam.qmui.widget.grouplist.QMUICommonListItemView.ACCESSORY_TYPE_CHEVRON;
import static com.qmuiteam.qmui.widget.grouplist.QMUICommonListItemView.HORIZONTAL;

import android.view.View;

import androidx.annotation.NonNull;

import com.qmuiteam.qmui.util.QMUIPackageHelper;
import com.qmuiteam.qmui.widget.dialog.QMUIDialog;
import com.qmuiteam.qmui.widget.dialog.QMUIDialogAction;
import com.qmuiteam.qmui.widget.grouplist.QMUICommonListItemView;
import com.qmuiteam.qmui.widget.grouplist.QMUIGroupListView;
import com.theone.music.R;
import com.theone.music.app.util.CacheUtil;
import com.theone.music.databinding.FragmentSettingBinding;
import com.theone.mvvm.base.viewmodel.BaseViewModel;
import com.theone.mvvm.ext.qmui.QMUICommonListItemViewExtKt;
import com.theone.mvvm.ext.qmui.QMUIDialogExtKt;
import com.theone.mvvm.ext.qmui.QMUIFragmentExtKt;

/**
 * @author The one
 * @date 2022-04-11 13:45
 * @describe 设置
 * @email 625805189@qq.com
 * @remark
 */
public class SettingFragment extends BaseFragment<BaseViewModel, FragmentSettingBinding> implements View.OnClickListener {

    private QMUICommonListItemView mVersion, mLoginOut;

    @Override
    public void initView(@NonNull View view) {
        QMUIGroupListView groupListView = getDataBinding().groupListView;
        mVersion = QMUICommonListItemViewExtKt.createItem(groupListView,
                "当前版本",
                "Ver " + QMUIPackageHelper.getAppVersion(mActivity),
                R.drawable.svg_setting_version, HORIZONTAL, ACCESSORY_TYPE_CHEVRON);

        QMUICommonListItemViewExtKt.addToGroup(groupListView, new QMUICommonListItemView[]{mVersion}, "关于", null, null);

        mLoginOut = QMUICommonListItemViewExtKt.createItem(groupListView,
                "退出账号",
                null,
                R.drawable.svg_setting_version, HORIZONTAL, ACCESSORY_TYPE_CHEVRON);

        if (CacheUtil.INSTANCE.isLogin()) {
            QMUICommonListItemViewExtKt.addToGroup(groupListView, new QMUICommonListItemView[]{mLoginOut}, "关于", null, this);
        }

    }


    @Override
    public void onClick(View v) {
        QMUIDialogExtKt.showMsgDialog(mActivity, "提示", "是否退出当前账号", "取消", "确定", new QMUIDialogAction.ActionListener() {
            @Override
            public void onClick(QMUIDialog dialog, int index) {
                dialog.dismiss();
                if (index > 0) {
                    getEventVm().setUserInfo(null);
                    CacheUtil.INSTANCE.loginOut();
                    QMUIFragmentExtKt.showSuccessTipsExitDialog(SettingFragment.this, "退出成功");
                }
            }
        }, QMUIDialogAction.ACTION_PROP_NEGATIVE);
    }


}

File: app/src/main/java/com/theone/music/ui/fragment/SettingFragment.kt
package com.theone.music.ui.fragment

import android.view.View
import com.qmuiteam.qmui.util.QMUIPackageHelper
import com.qmuiteam.qmui.widget.dialog.QMUIDialogAction
import com.qmuiteam.qmui.widget.grouplist.QMUICommonListItemView
import com.theone.music.R
import com.theone.music.app.util.CacheUtil
import com.theone.music.databinding.FragmentSettingBinding
import com.theone.music.viewmodel.EventViewModel
import com.theone.mvvm.base.fragment.BaseQMUIFragment
import com.theone.mvvm.base.viewmodel.BaseViewModel
import com.theone.mvvm.core.base.fragment.BaseCoreFragment
import com.theone.mvvm.ext.getAppViewModel
import com.theone.mvvm.ext.qmui.*

//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛
/**
 * @author The one
 * @date 2022-04-07 16:19
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */
class SettingFragment :BaseCoreFragment<BaseViewModel,FragmentSettingBinding>(),
    View.OnClickListener{

    private val mEvent: EventViewModel by lazy { getAppViewModel<EventViewModel>() }

    private lateinit var mLoginOut: QMUICommonListItemView
    private lateinit var mVersion: QMUICommonListItemView

    override fun getRootBackgroundColor(): Int = R.color.qmui_config_color_background

    override fun initView(root: View) {
        setTitleWitchBackBtn("设置")
        getDataBinding().groupListView.run {
            mVersion = createItem(
                "当前版本",
                "Ver " + QMUIPackageHelper.getAppVersion(mActivity),
                R.drawable.svg_setting_version
            )
            addToGroup(mVersion, title = "关于", listener = this@SettingFragment)

            mLoginOut = createItem("退出账号", drawable = R.drawable.svg_setting_login_out)
            if (CacheUtil.isLogin()) {
                addToGroup(mLoginOut, listener = this@SettingFragment, title = "")
            }
        }
    }

    override fun createObserver() {

    }

    override fun onClick(v: View?) {
        when (v) {
            mLoginOut -> {
                showLoginOutDialog()
            }
        }
    }

    private fun showLoginOutDialog() {
        context?.showMsgDialog(
            "提示", "是否退出当前账号",
            listener = QMUIDialogAction.ActionListener { dialog, index ->
                dialog.dismiss()
                if (index > 0) {
                    mEvent.setUserInfo(null)
                    CacheUtil.loginOut()
                    showSuccessTipsExitDialog("退出成功")
                }
            }, prop = QMUIDialogAction.ACTION_PROP_NEGATIVE
        )
    }

}